sdk/storage/azure_storage_blob/src/parsers.rs

```diff
@@ -35,3 +39,49 @@ pub fn format_page_range(offset: u64, length: u64) -> Result<String, Error> {
     let content_range = format!("bytes={}-{}", offset, end_range);
     Ok(content_range)
 }
+
+/// Takes in a HashMap of blob tags and serializes them into the `BlobTags` model.
+///
+/// # Arguments
+///
+/// * `tags` - A hash map containing the name-value pairs associated with the blob as tags.
+pub(crate) fn serialize_blob_tags(tags: HashMap<String, String>) -> BlobTags {
+    let mut blob_tags = vec![];
+
+    for (k, v) in tags.into_iter() {
+        let blob_tag = BlobTag {
+            key: Some(k),
+            value: Some(v),
+        };
+        blob_tags.push(blob_tag);
+    }
+    BlobTags {
+        blob_tag_set: Some(blob_tags),
+    }
+}
+
+/// Takes in a `get_tags()` response and deserializes the `BlobTags` model into a HashMap of blob tags.
+///
+/// # Arguments
+///
+/// * `response` - The `get_tags()` response to be deserialized.
+pub(crate) async fn deserialize_blob_tags(
+    response: Response<BlobTags, XmlFormat>,
+) -> azure_core::Result<Response<HashMap<String, String>, XmlFormat>>
+where
+{
+    let mut blob_tags_map = HashMap::new();
+    let status = response.status();
+    let headers = response.headers().clone();
+    let blob_tags = response.into_body().await?;
+
+    if let Some(blob_tag_set) = blob_tags.blob_tag_set {
+        for blob_tag in blob_tag_set {
+            blob_tags_map.insert(blob_tag.key, blob_tag.value);
+        }
+    }
+
+    let xml_body = xml::to_xml_with_root("HashMap", &blob_tags_map)?;
```

Comment:
This feels pretty hacky to put it back into XML. I'm not sure if `Response` has a mechanism for us to say, we've already desterilized, here is the result so when someone calls `into_body` it just gives that back. Or perhaps there is a different way to transform this response with adding a custom deserialization method? Not sure.